PhabricatorBMOAuthAdapter.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+final class PhabricatorBMOAuthAdapter extends PhutilAuthAdapter {
+
+  private $account_id;
+  private $account_email;
+  private $account_name;
+  private $account_real_name;
+  private $auth_uri;
+
+  private $type;
+  private $domain;
+
+  public function getAccountID() {
+    return $this->account_id;
+  }
+
+  public function setAccountId($id) {
+    $this->account_id = $id;
+  }
+
+  public function getAdapterType() {
+    return $this->type;
+  }
+
+  public function setAdapterType($type) {
+    $this->type = $type;
+  }
+
+  public function getAdapterDomain() {
+    // Unfortunately a string default must be provided here.
+    // The auth provider will fail to initially register if not provided,
+    // even when the custom "domain" field is provided because
+    // getAdapterDomain() returns null
+
+    return $this->domain ?: 'bugzilla.mozilla.org';
```

Comment:
I'll also note two things:

1.  Everything works with the providerDomain being the bugzilla.mozilla.org address
2.  I'm still working on a way to set that value reliably.